packages/integration-sdk-runtime/src/execution/step.ts

```diff
@@ -39,16 +42,31 @@ export async function executeSteps<
   graphObjectStore: GraphObjectStore;
   createStepGraphObjectDataUploader?: CreateStepGraphObjectDataUploaderFunction;
   beforeAddEntity?: BeforeAddEntityHookFunction<TExecutionContext>;
+  dependencyGraphOrder?: string[];
 }): Promise<IntegrationStepResult[]> {
-  return executeStepDependencyGraph({
-    executionContext,
-    inputGraph: buildStepDependencyGraph(integrationSteps),
-    stepStartStates,
-    duplicateKeyTracker,
-    graphObjectStore,
-    createStepGraphObjectDataUploader,
-    beforeAddEntity,
-  });
+
+  const stepsByGraphId = seperateStepsByDependencyGraph(integrationSteps);
+  let allStepResults: IntegrationStepResult[] = [];
+
+  for (const graphId of [
+    DEFAULT_DEPENDENCY_GRAPH_IDENTIFIER,
+    ...(dependencyGraphOrder ?? []),
+  ]) {
+    const steps = stepsByGraphId[graphId] ?? [];
```

Comment:
I'm wondering if we should warn if a specific `dependencyGraphOrder` is referenced in the invocation config that is not referenced by any step. Not an error, but at least some type of indicator that something _could_ be wrong here.

Reply:
Yeah, I'll add that real quick.